iterator.js

```diff
@@ -14,11 +16,32 @@ function Iterator (db, options) {
 
 util.inherits(Iterator, AbstractIterator)
 
-Iterator.prototype.seek = function (key) {
-  if (typeof key !== 'string')
-    throw new Error('seek requires a string key')
+Iterator.prototype.seek = function (target) {
+  if (this._ended)
+    throw new Error('cannot call seek() after end()')
+  if (this._nexting)
+    throw new Error('cannot call seek() before next() has completed')
+  if (typeof target !== 'string' && !Buffer.isBuffer(target))
+    throw new Error('seek() requires a string or buffer target')
+
+  // Check if target is cached
+  var length = this.cache ? this.cache.length : 0
+
+  if (length) {
+    if (typeof target === 'string') {
+      target = new Buffer(target)
+    }
+
+    for(var i=length-1; i>0; i-=2) {
+      var c = compare(this.cache[i], target)
+      if (this.reverse ? c <= 0 : c >= 0) {
+        return void this.cache.splice(i+1, length)
```

Comment:
whitespace, `i + `

also, `return void`? a bit too clever, probably needs multiple lines to achieve this